### ISSUE TITLE
📜 Allow for extra content fro some languages?

### ISSUE THREAD

**Issue by Felienne** [hedyorg/hedy#5721]
In addition to #5720, the Chinese content makes would like to create some additional pages for their content, such as:

> The main content will include:
> +Community introduction
> +Hedy learning documents
> +Hedy promotional event announcements/notices/reports
> +Hedy training resources, appointments

I am totally ok with that suggestion, but I am not sure how to practically make that happen, because currently all content exists in all languages and we don’t have support for content only in one language.

**Comment by Felienne** [comment]
Note from @ZoomQuiet in #5720 that is relevant here:

> Moreover, although using hostnames is simpler and easier to spread than using parameters in terms of URL, 
> we further envision using cn.hedy.org as the official Chinese community. 
> In addition to being able to directly publish all Chinese content on the official website, 
> we also hope to be able to publish information from each community on our own, such as: 
> 1. Various activities organized in China cn.hedy/org/events/ 
> 1. Various notes written for China cn.hedy/org/notes/ 
> 1. Voice/video/... materials recorded in the Chinese community cn.hedy/org/sources/ 
> 1. ... 
> 
> 
> Therefore, official technical cooperation may be required: 
> Use rules such as Nginx to replace all links starting with https://hedy.org/hedy?lang=cn with https://cn.hedy.org/doc, 
> and then point cn.hedy.org to the official Chinese community website we are preparing to release. 
> We generally use the github-pages service to publish, 
> and may need to point the domain name's CNAME to a .github.io domain name.
> 
>

**Comment by Felienne** [comment]
Just to clarify, the only reason to not do this straight away is that it will be somewhat complex to build, so we need to figure out how to do this properly (such that is can work with existing tools and works for non-technical people too across languages) it is not a matter of not wanting you to have this option.

A very very easy way to get this done quickly is that you setup your own website somewhere (like github pages) and we simply link to it labeling it the official Chinese community site (through something like cn.hedy.org/community) Let me know if that works!

**Comment by ZoomQuiet** [comment]
> ...
> 
> A very very easy way to get this done quickly is that you setup your own website somewhere (like github pages) and we simply link to it labeling it the official Chinese community site (through something like cn.hedy.org/community) Let me know if that works!

This is currently feasible, 

but for readers, it feels a bit subtle, 
- just like you originally wanted to visit google.com, 
- but suddenly jumped to google.github.io, 
- psychologically, you may not trust the new domain name. 

For this reason, we may need to purchase domain names such as `cnhedy.org` so that readers can feel as close as possible; 

of course, linking from the official website to the Chinese community website is an officially recognized behavior, 
which is absolutely feasible; 

please tell us about hedy's decision so that we can grab the domain name in time?

**Comment by Felienne** [comment]
> This is currently feasible,

Great, let's explore that path more!

> but for readers, it feels a bit subtle,
> 
> * just like you originally wanted to visit google.com,
> * but suddenly jumped to google.github.io,
> * psychologically, you may not trust the new domain name.

Yes, I see! I am sure though that there are methods to keep the original domain name in the address bar while doing a redirect if both domains fiddle with the settings a bit. Can you look into how that could work to increase trust?

**Comment by Felienne** [comment]
Thanks for the detailed info! We will discuss this in our next contributors meeting on Tuesday (feel free to drop in if you want/can (2PM CET)

**Comment by ZoomQuiet** [comment]
> Thanks for the detailed info! We will discuss this in our next contributors meeting on Tuesday (feel free to drop in if you want/can (2PM CET)

i'll try join, but can not make sure now.

thanx for all,

this issue, can usage all others country's community support.

**Comment by jpelay** [comment]
Hello @ZoomQuiet! We were debating about this issue in the last couple of meetings and came to a proposal for you guys!

We can provide you with a HTML template that you can edit as you'd like. The link to access this template would be something like: cn.hedy.org/community In order to edit this template you would upload PRs to our repo which in turn would be approved by us.

Does this proposal work for you?

**Comment by ZoomQuiet** [comment]
> Hello @ZoomQuiet! We were debating about this issue in the last couple of meetings and came to a proposal for you guys!
> 
> We can provide you with a HTML template that you can edit as you'd like. The link to access this template would be something like: cn.hedy.org/community In order to edit this template you would upload PRs to our repo which in turn would be approved by us.
> 
> Does this proposal work for you?

No problem,
My understanding is:

- 1: The Chinese community can use cn.hedy.org as the domain name to publish community activities and other content other than official documents
- 2: However, it is necessary to use the same webpage template as the official website

So, how will such webpages be published?
Or should I use my previous solution?
Can I use various services like ghithub-pages to publish them in other domain names,
After being reviewed by hedy.org,
then use a reverse proxy like nginx to forward them to the cn.hedy.org domain name?

**Comment by jpelay** [comment]
> No problem, My understanding is:
> 
> * 1: The Chinese community can use cn.hedy.org as the domain name to publish community activities and other content other than official documents
> * 2: However, it is necessary to use the same webpage template as the official website


No, it still would be our site, so no Ngix redirection required. The subdomain would point to a html file within our repo that will be handled by you.

**Comment by ZoomQuiet** [comment]
@jpelay 
> ...No, it still would be our site, so no Ngix redirection required. The subdomain would point to a html file within our repo that will be handled by you.

To clarify the details, my current understanding is:

- cn.hedy.org will be shared with the Chinese community responsible for the content
- However, there is only one html file that can be freely edited?

The good news is that we will be allowed to maintain content based on .md through a mechanism similar to github-pages?

So, since git repositories are used,
I suggest:

- The webpages pointed to by cn.hedy.org are compiled independently through sub-repositories
- In this way, the Chinese community does not have to worry about and maintain the same template files as the main site, and can just write .md
- Furthermore, this can also allow the Chinese community to create different directories for long-term collection and publication of activities and achievements from all over the world

That is, allow the Chinese community to publish a group of pages with structural relationships to each other in the cn.hedy.org domain name, rather than a single page?

**Comment by Felienne** [comment]
> @jpelay
> 
> > ...No, it still would be our site, so no Ngix redirection required. The subdomain would point to a html file within our repo that will be handled by you.
> 
> To clarify the details, my current understanding is:
> 
> * cn.hedy.org will be shared with the Chinese community responsible for the content
> * However, there is only one html file that can be freely edited?
> 
> The good news is that we will be allowed to maintain content based on .md through a mechanism similar to github-pages?
> 
> So, since git repositories are used, I suggest:
> 
> * The webpages pointed to by cn.hedy.org are compiled independently through sub-repositories
> * In this way, the Chinese community does not have to worry about and maintain the same template files as the main site, and can just write .md
> * Furthermore, this can also allow the Chinese community to create different directories for long-term collection and publication of activities and achievements from all over the world
> 
> That is, allow the Chinese community to publish a group of pages with structural relationships to each other in the cn.hedy.org domain name, rather than a single page?

Hi @ZoomQuiet!

I understand the question, but just to get things going, can we start with, instead of 1 HTML page, a folder of HTML pages? Than you can indeed make different pages and link to each other, but we don't have to organize subrepos (which will be a large hassle for a lot of languages). Please understand that we have to balance your needs with the option that we will have to maintain it 50 fold.

@jpelay Can you make a demo with a few HTML pages instead of one? If we have a concrete example, the discussion might be eased.

**Comment by ZoomQuiet** [comment]
@Felienne Agree, a set of html or a specified folder,
are the minimum viable start;
However, I want to remind you, how to ensure that the style of the webpages published here is consistent with the official one?

If there is no such requirement, we are used to using Python tools,
automatically compiling pure html webpages from markdown,
such as MkDocs https://www.mkdocs.org/

@jpelay 
However, if you want the webpages published by the Chinese community to have the same style as the official website,
you may have to provide corresponding conversion tools;

**Comment by Felienne** [comment]
> @Felienne Agree, a set of html or a specified folder, are the minimum viable start; 

Let's work from there, @jpelay has made a preliminary PR, please check that out: https://github.com/hedyorg/hedy/pull/5814

>However, I want to remind you, how to ensure that the style of the webpages published here is consistent with the official one?

They will be served using the same css, so in that sense, it will look and feel the same. Plus we (you) will add new content through PRs, so we as core team can always keep an eye on what thing look like. 

> If there is no such requirement, we are used to using Python tools, automatically compiling pure html webpages from markdown, such as MkDocs https://www.mkdocs.org/

Yeah, I am sorry but you will have to change your workflow a bit to accommodate this solution, which I know might be annoying but I hope you understand we have many contradicting constraints to work with!
 
> @jpelay However, if you want the webpages published by the Chinese community to have the same style as the official website, you may have to provide corresponding conversion tools;

Here are there there might be a small drift, (ie in terms of what headers to use) and we can live with that. Hope you can too!
